sanity/doctors/Diagnosis.js

```diff
@@ -6,7 +6,7 @@ export class Diagnosis {
     this.log = new Log(this.constructor.name)
   }
 
-  async doTreatment() {
+  async doTreatment(fromBlock) {
```

Comment:
We're already supplying `fromBlock` to the prepare method, don't you think it might be a good idea to have it as instance variable?